src/navsat_transform.cpp

```diff
@@ -130,14 +131,14 @@ NavSatTransform::NavSatTransform(const rclcpp::NodeOptions & options) :
   }
 
   odom_sub_ = this->create_subscription<nav_msgs::msg::Odometry>(
-    "odometry/filtered", 1, std::bind(&NavSatTransform::odomCallback, this, _1));
+    "odometry/filtered", rclcpp::SensorDataQoS(rclcpp::QoSInitialization(RMW_QOS_POLICY_HISTORY_KEEP_LAST, queue_size_)), std::bind(&NavSatTransform::odomCallback, this, _1));
```

Comment:
Cleaner way https://github.com/ros-planning/navigation2/blob/master/nav2_costmap_2d/plugins/obstacle_layer.cpp#L181

```
    rmw_qos_profile_t custom_qos_profile = rmw_qos_profile_sensor_data;
    custom_qos_profile.depth = queue_size;
    ...
    create_sub<>(..., SensorDataQoS(custom_qos_profile), ...);
```

You can make 1 profile and just update the depth for each of the subscribers.

Reply:
Thanks, much nicer. Actually did it this way somewhere else, but for some reason I did it this way instead. 

Hm ok, take a look at how I did, it's a bit different the navsat_transform node and the ros_filter.cpp. Sorry I don't think I fully got what you meant :/ 

In navsat_transform my guess was that I could use the same profile for each, and the queue_size is the same for all. 

For ros_filter, I could do it the way you described, just perhaps create one profile early and before each subscription set the `depth` with `queue_size`  I guess it will not reallocate, which saves memory and speeds it up a tiny bump, on the other hand, I was trying to get google the programming principle of keeping things together that is used together haha but I couldn't find it. My concern was that putting the profile creation early early and then just reassigning the depth, that it might be very far from where it's used. But maybe I'm just overthinking this and that it will be a smoother experience (if that is what you meant).

Sorry for going back and forth on this one, but since we've kept at it this long, we might as well get it right, and I'm learning a lot :D at least I think so :P

Reply:
Your suggestion to use `from_rmw` seems fine. I want to drop the `RMW_QOS_POLICY_HISTORY_KEEP_LAST` because 

1) I'm not sure what effects that has
2) It might actually change the sensor data QoS too much to not connect correctly

https://github.com/ros-planning/navigation2/blob/master/nav2_costmap_2d/plugins/voxel_layer.cpp#L84

Another navigation example where we use a custom QoS for a publisher.

Reply:
Ok thanks, I was also worried about 2. or at least wondering what in the profile that actually made it work. Ok i'll use from_rmw. 

I just moved back from NYC to SF, so haven't been able to get to this, but will fix it asap so that we can get this merged :)

Reply:
OK this is now fixed, lemme know what you think. I went for this https://github.com/ros-planning/navigation2/blob/master/nav2_costmap_2d/plugins/voxel_layer.cpp#L84 approach. It's pretty explicit and short. Only downside is it's not clear that it's the sensor profile (with depth 1 for some though).
@Jconn could you confirm that these are similar to what you already did and worked?